File: algorithms/1_easy/C/01995_count_special_quadruplets/bruteforce.cpp
class Solution {
public:
    int countQuadruplets(vector<int>& nums) {
        const size_t n{nums.size()};
        int count{0};

        for (size_t d{n - 1}; d < n; --d) {
            for (size_t c{d - 1}; c < d; --c) {
                for (size_t b{c - 1}; b < c; --b) {
                    for (size_t a{b - 1}; a < b; --a) {
                        if (nums[d] == nums[a] + nums[b] + nums[c]) {
                            count += 1;
                        }
                    }
                }
            }
        }

        return count;
    }
};

File: algorithms/1_easy/C/01995_count_special_quadruplets/bruteforce.rs
impl Solution {
    pub fn count_quadruplets(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut count = 0_i32;

        for d in (0..n).rev() {
            for c in (0..d).rev() {
                for b in (0..c).rev() {
                    for a in (0..b).rev() {
                        if nums[d] == nums[a] + nums[b] + nums[c] {
                            count += 1;
                        }
                    }
                }
            }
        }

        count
    }
}
